### ISSUE TITLE
brownie is not aware of changing implementation and new abi of a proxy contract

### ISSUE THREAD

**Issue by gosuto-inzasheru** [eth-brownie/brownie#1314]
### Environment information

* `brownie` Version: 1.17.0
* `ganache-cli` Version: 6.12.2 (core v2.13.2)
* ~`solc` Version: x.x.x~
* Python Version: 3.7.6
* OS: macos

### What was wrong?

when introducing a new `Contract()` to brownie when having `autofetch_source: True`, it runs a `from_explorer` and gets necessary abi. however, if that contract happens to be a proxy, brownie is never aware of any changes to the implementation contract.

in other words, if in the next implementation of the contract the signature of the contract changes (eg a new func is introduced), brownie never becomes aware, and will raise a
> AttributeError: Contract 'SomeProxyContract' object has not attribute 'newFunc'

### How can it be fixed?

Pointers to possible solutions or responsible code are welcome :)
